src/blob_file_iterator.cc

```diff
@@ -120,6 +136,10 @@ void BlobFileIterator::IterateForPrev(uint64_t offset) {
     status_ = decoder_.DecodeHeader(&header_buffer);
     if (!status_.ok()) return;
     total_length = kRecordHeaderSize + decoder_.GetRecordSize();
+    iterate_offset_ += total_length;
+    if (block_size_ != 0) {
+      total_length += AdjustOffsetToNextBlockHead();
```

Comment:
Should we put this IF condition before line 139 (`iterate_offset_ += total_length;`)? I'm assuming we want the next `iterate_offset_ ` to be the next block head. 

Also, on line 145 we do `iterate_offset_ -= total_length` which I believe is to get the previous offset. If we increase total_length after line 139, it seems to me that we'll get the wrong offset on line 145. Am I missing something?

Reply:
`AdjustOffsetToNextBlockHead ()` modifies `iterate_offset_` internally.
`AdjustOffsetToNextBlockHead ()` is used in multiple places, so the adjustment of `iterate_offset_` better stays inside of the function. Here, I want to reuse the same code piece as much as possible, so it might be a little bit counter-intuitive